src/rules/whitespaceRule.ts

```diff
@@ -306,7 +320,16 @@ function walk(ctx: Lint.WalkContext<Options>) {
                 }
         }
     });
-
+    function matchPostblockPreKeyword(node: ts.Node): boolean {
+        const parent = node.parent!;
+        const isDoWhile = utils.isDoStatement(parent);
+        const isIfElse = utils.isIfStatement(parent)
+            && parent.elseStatement !== undefined
+            && parent.elseStatement !== node;
+        const isTryCatchFinally = utils.isTryStatement(parent)
+            && (parent.tryBlock === node || parent.catchClause === node);
```

Comment:
You can use the following to avoid the additional `case` for `CatchClause` above (sorry my initial suggestion got that part wrong):
```ts
const isTryCatchFinally = utils.isTryStatement(parent) && parent.tryBlock === node ||
    utils.isCatchClause(parent) && parent.parent!.finallyBlock !== undefined;
```

Reply:
Done